### PR TITLE
[13.0] Add module l10n_ch_qr_no_amount

### PR DESCRIPTION
**Pull request by grindtildeath**
This module allows to print QR bills without any amount, so that the customer can define the amount to pay.

Not setting the amount allows to use the same QR code for multiple payments referencing the same invoice.

### CODE REVIEW

**Review by mmequignon — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Comment by ajaniszewska-dev** [comment]
@grindtildeath  fix pre-commit

File: l10n_ch_qr_no_amount/models/res_partner_bank.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        self, amount, currency_name, debtor_partner, reference_type, reference, comment
+    ):
+        res = super()._prepare_swiss_code_url_vals(
+            amount, currency_name, debtor_partner, reference_type, reference, comment
+        )
+        if self.env.context.get("_no_amount_qr_code", False):
+            res = self._remove_swiss_code_amount(res)
+        return res
+
+    def _remove_swiss_code_amount(self, values_list):
```
Comment (TDu) [inline l10n_ch_qr_no_amount/models/res_partner_bank.py:20]:
```suggestion
    @api.model
    def _remove_swiss_code_amount(self, values_list):
```

File: l10n_ch_qr_no_amount/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This module allows to print QR bills without any amount, so that the customer can define the amount to pay.
+
+Not setting the amount allows to use the same QR code for multiple payments referencing the same invoice.
```
Comment (TDu) [inline l10n_ch_qr_no_amount/readme/DESCRIPTION.rst:4]: Maybe could be stated that how it is done ? with the context key ?

**Review by TDu — COMMENTED** [review]
Do we want a unit test ?

File: l10n_ch_qr_no_amount/models/res_partner_bank.py (hunk 1 above)
Comment (grindtildeath) [reply l10n_ch_qr_no_amount/models/res_partner_bank.py:20]: As it's not done in the super methods, I guess we can avoid adding it if unnecessary:
https://github.com/odoo/odoo/blob/13.0/addons/l10n_ch/models/res_bank.py#L192
https://github.com/odoo/odoo/blob/13.0/addons/l10n_ch/models/res_bank.py#L264

File: l10n_ch_qr_no_amount/readme/DESCRIPTION.rst (hunk 2 above)
Comment (grindtildeath) [reply l10n_ch_qr_no_amount/readme/DESCRIPTION.rst:4]: I added a bit of explanation how to use it. I don't think it makes sense to explain the technical side in the module's description.

**Comment by grindtildeath** [comment]
@TDu I guess we don't need a unit test to only check if a value was removed from the list or do we?

**Comment by TDu** [comment]
@grindtildeath  Thank you, LG

**Pull request merged**